Project/VetAppointmentAPI/VetAppointmentAPI/VetAppointment.WebAPI/Controllers/OfficesController.cs

```diff
@@ -13,63 +18,55 @@ namespace VetAppointment.WebAPI.Controllers
         {
             this.officeRepository = officeRepository;
-            this.officeValidator = validator;
-            this.mapper = mapper;
+            this.mediator = mediator;
         }
 
         // GET: api/<OfficesController>
         [HttpGet]
-        public async Task<IActionResult> Get()
+        public async Task<List<OfficeResponse>> Get()
         {
-            return Ok(await officeRepository.All());
+            return await mediator.Send(new GetAllOfficesQuery());
         }
 
         // GET api/<OfficesController>/5
         [HttpGet("{id}")]
-        public async Task<IActionResult> Get(Guid id)
+        public async Task<ActionResult<OfficeResponse>> Get(Guid id)
         {
-            var office = await officeRepository.Get(id);
-            if (office == null)
+            var result = await mediator.Send(new GetOfficeByIdQuery
             {
-                return NotFound();
-            }
-            return Ok(mapper.Map<OfficeDto>(office));
+                Id = id
+            });
+            return Ok(result);
         }
 
         // POST api/<OfficesController>
         [HttpPost]
-        public async Task<IActionResult> Post([FromBody] OfficeDto officeDto)
+        public async Task<ActionResult<OfficeResponse>> Post([FromBody] CreateOfficeCommand command)
         {
-            var validation = officeValidator.Validate(officeDto);
-            if (!validation.IsValid || officeDto.Address == null)
-                return StatusCode(400, validation.Errors.First().ErrorMessage);
-            Office office = mapper.Map<Office>(officeDto);
-            await officeRepository.Add(office);
-            await officeRepository.SaveChanges();
-
-            return Created(nameof(Office), mapper.Map<OfficeDto>(office));
+            var result = await mediator.Send(command);
+            return Ok(result);
         }
 
         // PUT api/<OfficesController>/5
-        [HttpPut]
-        public async Task<IActionResult> Put([FromBody] OfficeDto officeDto)
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Put([FromRoute] Guid id, [FromBody] UpdateOfficeCommand command)
         {
-            var validation = officeValidator.Validate(officeDto);
-            if (!validation.IsValid)
-                return StatusCode(400, validation.Errors.First().ErrorMessage);
-            Office? office = await officeRepository.Get(officeDto.OfficeId);
-            if (office == null)
-                return NotFound();
-            mapper.Map(officeDto, office);
-            officeRepository.Update(office);
-            await officeRepository.SaveChanges();
+            var drug = await mediator.Send(new GetOfficeByIdQuery
+            {
+                Id = id
+            });
+
+            if (drug == null)
+            {
+                return NotFound($"Drug with id: {id} was not found");
+            }
 
+            var result = await mediator.Send(command);
             return NoContent();
```

Comment:
Same here, does the mediator return early if there are errors? Otherwise there need to be explicit return points.